### PR TITLE
test: run tests of all go.mods

### PR DESCRIPTION
**Pull request by crodriguezvega**
## Description

Trying to see if `make test` could run all tests. [Taken from SDK](https://github.com/cosmos/cosmos-sdk/blob/main/Makefile#L227-L253).

closes: #XXXX

### Commit Message / Changelog Entry

```bash
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/docs/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/callbacks/go.mod [hunk 1]
```diff
@@ -17,6 +17,8 @@ require (
 	github.com/cosmos/ibc-go/modules/capability v1.0.0-rc1
 	github.com/cosmos/ibc-go/v7 v7.2.0
 	github.com/spf13/cast v1.5.1
+	github.com/spf13/cobra v1.7.0
```
Comment (crodriguezvega) [inline modules/apps/callbacks/go.mod:20]: This must have changed because I run `go mod tidy`...

File: e2e/tests/core/02-client/client_test.go [hunk 2]
```diff
@@ -1,3 +1,6 @@
+//go:build !test_e2e
+// +build !test_e2e
```
Comment (crodriguezvega) [inline e2e/tests/core/02-client/client_test.go:2]: Added this to e2e tests, so that we can skip them.

File: Makefile [hunk 3]
```diff
@@ -205,11 +205,32 @@ check-test-unit-amino: ARGS=-tags='ledger test_ledger_mock test_amino'
...
+	@echo "Starting unit tests"; \
+	finalec=0; \
+	for module in $(SUB_MODULES); do \
+		cd ${CURRENT_DIR}/$$module; \
+		echo "Running unit tests for $$(grep '^module' go.mod)"; \
+		go test -mod=readonly $(ARGS) $(EXTRA_ARGS) $(TEST_PACKAGES) ./... ; \
+		ec=$$?; \
+		if [ "$$ec" -ne '0' ]; then finalec=$$ec; fi; \
+	done; \
+	exit $$finalec
```
Comment (crodriguezvega) [inline Makefile:233]: Both blocks look the same to me, but SDK had it the same way...

Comment (DimitrisJim) [reply Makefile:233]: the diff is in piping to `tparse`. I didn't like `tparse` when I tried it. We could just remove the conditional unless someone else uses it.

**Comment by DimitrisJim** [comment]
guess this is only applicable to local running of files. `test.yml` would need to be tweaked in a similar way. I was pretty sure SDK used a script for both Makefile && github runner but I could be mistaken.

**Comment by DimitrisJim** [comment]
Marking this as R4R, locally, tests all run okay for me.

**Review by chatton — COMMENTED** [review]
This will be super useful! I do have one suggestion though, I think it might be more maintainable to just keep this as a separate `.sh` script and just run it from the make file. I think this will be more manageable overall.

**Review by charleenfei — APPROVED** [review]
thank you so much for this @crodriguezvega! super useful. single commits for linting post push -- 

(also works locally for me)

**Review by chatton — APPROVED** [review]
LGTM Thank you @crodriguezvega  and @DimitrisJim 

This will be super useful 💪

**Comment by crodriguezvega** [comment]
Thank you @DimitrisJim for taking this to the finish line!

**Pull request merged**